apps/sync/js/adapters/history.js

```diff
@@ -168,12 +189,66 @@ var HistoryHelper = (() => {
     });
   }
 
+  function checkIfCleared(userid) {
+    return getLastRevisionId(userid).then(lastRevisionId => {
+      return _ensureStore().then(store => {
```

Comment:
We are using `getLastRevisionId` only here and this method is also returning the last revision ID, so it may make sense to move all this code into `getLastRevisionId` and get rid of `checkIfCleared`.